pkg/test/step.go

```diff
@@ -533,6 +566,11 @@ func (s *Step) LoadYAML(file string) error {
 			} else {
 				return fmt.Errorf("failed to load TestAssert object from %s: it contains an object of type %T", file, obj)
 			}
+
+			s.Programs, err = expressions.LoadPrograms(s.Assert)
+			if err != nil {
+				return fmt.Errorf("failed to load programs: %w", err)
```

Comment:
The term "programs" are not mentioned in the API. We need to formulate error messages in terms that appear in the API. However this function processes both expressions and resourceRefs, so and I cannot think of a common term to describe them. If you cannot either, then perhaps this should propagate the error without wrapping, and `LoadPrograms` should take care of formulating a user-friendly error message... 🤔